app/controllers/index.js

```diff
@@ -54,11 +56,28 @@ export default class IndexController extends Controller {
     });
   }
 
+  async fetchAdditionalUserDetails(users) {
+    let userPromises = [];
+    users.forEach((user) => {
+      userPromises.push(fetch(user.url).then((response) => response.json()));
+    });
+
+    let userDataResponses = await all(userPromises);
+
+    return users.map((user, index) => ({
+      ...user,
+      name: userDataResponses[index].name || "",
```

Comment:
**suggestion (non-blocking):**

I think we can avoid setting a default value here (and let the consuming site default it however it needs to). I wonder if `name` is always defined.

```suggestion
      name: userDataResponses[index].name,
```